rrtmgp/kernels/mo_reorder_kernels.F90

```diff
@@ -44,18 +60,35 @@ subroutine reorder_123x321_kernel(d1, d2, d3, array_in, array_out) &
     real(wp), dimension(d1, d2, d3), intent( in) :: array_in
     real(wp), dimension(d3, d2, d1), intent(out) :: array_out
 
-    integer :: i1, i2, i3
 
-    !$acc parallel loop collapse(3) &
-    !$acc&     copyout(array_out(:d3,:d2,:d1)) &
-    !$acc&     copyin(array_in(:d1,:d2,:d3))
-    do i1 = 1, d1
-      do i2 = 1, d2
-        do i3 = 1, d3
-          array_out(i3,i2,i1) = array_in(i1,i2,i3)
+    integer :: i1, i2, i3, i10, i30, i1diff, i3diff, idiff
+    integer, parameter :: tile = 32
+
+    !$acc parallel vector_length(tile*tile) &
```

Comment:
As above - please comment to explain what's going on.

Reply:
That's basically the same as I answered above